addons/popochiu/editor/main_dock/popochiu_row/object_row/popochiu_object_row.gd

```diff
@@ -0,0 +1,363 @@
+@tool
+extends "res://addons/popochiu/editor/main_dock/popochiu_row/popochiu_row.gd"
+## Row for the main object types: Room, Character, Inventory item, Dialog
+
+enum Options {
+	DELETE = MenuOptions.DELETE,
+	ADD_TO_CORE,
+	CREATE_STATE_SCRIPT,
+}
+
+const DELETE_MESSAGE = "This will remove the [b]%s[/b] object in [b]%s[/b] scene. Uses of this \
+object in scripts will not work anymore. This action cannot be reversed. Continue?"
+const DELETE_ASK_MESSAGE = "Do you want to delete the [b]%s[/b] folder too?%s (cannot be reversed)"
+const ADD_TO_CORE_ICON = preload(
+	"res://addons/popochiu/editor/main_dock/popochiu_row/images/add_to_core.png"
+)
+
+@onready var btn_open: Button = %BtnOpen
+@onready var btn_script: Button = %BtnScript
+@onready var btn_state: Button = %BtnState
+@onready var btn_state_script: Button = %BtnStateScript
+
+
+#region Godot ######################################################################################
+func _ready() -> void:
+	# Assign icons
+	btn_open.icon = get_theme_icon("InstanceOptions", "EditorIcons")
+	btn_script.icon = get_theme_icon("Script", "EditorIcons")
+	btn_state.icon = get_theme_icon("Object", "EditorIcons")
+	btn_state_script.icon = get_theme_icon("GDScript", "EditorIcons")
+	
+	# Connect to signals and create the options for the menu
+	super()
+	
+	btn_open.pressed.connect(_open)
+	btn_script.pressed.connect(_open_script)
+	btn_state.pressed.connect(_edit_state)
+	btn_state_script.pressed.connect(_open_state_script)
+
+
+#endregion
+
+#region Virtual ####################################################################################
+func _get_state_template() -> Script:
+	return null
+
+
+func _get_location() -> String:
+	return "Popochiu"
+
+
+#endregion
+
+#region Public #####################################################################################
+func show_add_to_core() -> void:
+	label.modulate.a = 0.5
+	menu_popup.set_item_disabled(menu_popup.get_item_index(Options.ADD_TO_CORE), false)
+
+
+func show_create_state_script_button() -> void:
+	btn_state_script.disabled = true
+	menu_popup.set_item_disabled(menu_popup.get_item_index(Options.CREATE_STATE_SCRIPT), false)
+
+
+func hide_create_state_script_button() -> void:
+	menu_popup.remove_item(menu_popup.get_item_index(Options.CREATE_STATE_SCRIPT))
```

Comment:
I am puzzled about what you're doing here.

Can you explain these three functions?

Reply:
`show_add_to_core()` is used to make the "Add to core" option disabled. Why does the function have that name? That's something we'd have to ask my past self, the idiot.

`show_create_state_script_button()` was used mostly in previous versions of the plugin before the addition of the State scripts. That function disables the gear button (⚙️) in the row that allows to open the State script of the object, and enables the option in the context  menu to create such script.

`hide_create_state_script_button()` just removes the "Create State script" option in the context menu of that row.

Reply:
Correction:

`show_add_to_code()` is actually used to make the "Add to core" option enabled in the context menu of the row. This is used to show devs that the object is in the project, but it is not inside the `popochiu_data.cfg` file neither its corresponding singleton (e.g. **R**, **C**, **I**, **D**). This is useful in cases when you move things from another project or delete an object without removing its files.

Reply:
Done.

I removed the `show_create_state_script_button()` and `hide_create_state_script_button()` functions since every object in Popochiu 2.0 alpha 1 and later already has a state script.

Regarding the `show_add_to_core()` function, I changed its name and added a comment explaining what it does.